minecraft/networking/types.py

```diff
@@ -120,6 +120,16 @@ def send(value, socket):
         socket.send(struct.pack('>i', value))
 
 
+class FixedPointInteger(Type):
+    @staticmethod
+    def read(file_object):
+        return Integer.read(file_object) / 32
```

Comment:
Ah, the test you added revealed something else I didn't notice before: in Python 2.7, `/` with two integer operands performs integer division (this was changed in Python 3 so that `//` is integer division and `/` is true division). For this line to return a float, you should change `32` to `32.0` (and/or convert the read `int` into a `float`).

Reply:
Alternatively, `from __future__ import division`